src/lib.rs

```diff
@@ -38,7 +38,9 @@ pub use self::slave::{Slave, SlaveId};
 mod codec;
 
 mod frame;
-pub use self::frame::{Address, FunctionCode, Quantity, Request, Response};
+pub use self::frame::{
+    Address, Exception, ExceptionResponse, FunctionCode, Quantity, Request, Response,
```

Comment:
I think its a bad idea to make `ExceptionResponse` public with its fields also public.

We should add at least 2 methods for the `ExceptionResponse`:
- `exception()` that returns the `Exception`
- `function_code()` that returns the `FunctionCode`

And we should also enable a simple creation of the `ExceptionResponse` with a `new` method, or a method on `Request` / `Response` maybe to create a `ExceptionResponse` from it.

Reply:
After thinking about it more, I think only `Exception` should be in the public API.

The user shouldn't have to know the existence of `ExceptionResponse`, the server should be able to build the appropriate `Response` type (either `Response` or `ExceptionResponse`) by the return of the `call` function.

See #237 for more context.